File: finagle-core/src/main/java/com/twitter/finagle/ServiceFactory.java
package com.twitter.finagle;

import com.newrelic.api.agent.weaver.MatchType;
import com.newrelic.api.agent.weaver.Weave;
import com.newrelic.api.agent.weaver.Weaver;
import com.newrelic.instrumentation.finagle.core.Utils;

@Weave(type = MatchType.BaseClass)
public class ServiceFactory<Req, Rep> {

	public Service<Req,Rep> toService() {
		Service<Req,Rep> service = Weaver.callOriginal();
//		return Utils.getServiceWrapper(service);
		return service;
	}
}

File: finagle-core-18.2.0/src/main/scala/com/twitter/finagle/ServiceFactory.scala
package com.twitter.finagle

import com.newrelic.api.agent.weaver.{Weaver,Weave}
import com.newrelic.instrumentation.finagle.core.ServiceUtils

@Weave(originalName="com.twitter.finagle.ServiceFactory")
class ServiceFactory_instrumentation[Req, Rep] {

    def toService: Service[Req, Rep] = {
        val service : Service[Req, Rep] = Weaver.callOriginal()
        // val serviceName : String = ServiceUtils.getServiceFactoryName(this)
        // if(serviceName != null) {
        //     ServiceUtils.getInstance().addServiceName(service,serviceName,null)
        // }
        service
    }
}